sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorClient.cs

```diff
@@ -0,0 +1,982 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.ServiceBus.Core;
+using Azure.Messaging.ServiceBus.Diagnostics;
+using Microsoft.Azure.Amqp;
+
+namespace Azure.Messaging.ServiceBus
+{
+    /// <summary>
+    ///   A client responsible for reading <see cref="ServiceBusMessage" /> from a specific entity
+    ///   as a member of a specific consumer group.
+    ///
+    ///   A consumer may be exclusive, which asserts ownership over associated partitions for the consumer
```

Comment:
nit:  Probably want to tweak the first paragraph and zap the rest, as the concepts are Event Hubs-specific.